tests/test_metadata_handling.py

```diff
@@ -0,0 +1,67 @@
+import unittest
+import yaml
+from shaker import helpers
+from nose.tools import raises
+
+class TestMetadataHandling(unittest.TestCase):
+
+    # Sample metadata with duplicates
+    _sample_metadata_duplicates = {
+        "dependencies": [
+            "git@github.com:test_organisation/test1-formula.git==v1.0.1",
+            "git@github.com:test_organisation/test1-formula.git==v1.0.1",
```

Comment:
These two should have a different version, shouldn't they? And they'd still be resolved....

Or is this not how it works?

@mattmb This issue (#14) was reported by you - can you remember if you'd put the exact same formula+version twice or did you have different versions in each line?

If that's the case then @niallcreech we might need to move this up a layer to around here https://github.com/ministryofjustice/salt-shaker/blob/handle-metadata-duplicates/shaker/resolve_deps.py#L145-L150 - might need to rethink how we do this :(

(Sorry this is turning into such a back and forth PR – I'm doing a terrible job of either reviewing everything in one go and of providing enough detail of the ticket in the first place.)